core/src/test/java/org/apache/iceberg/TestFixedSizeSplitScanTaskIterator.java

```diff
@@ -24,9 +24,9 @@
 import org.junit.Assert;
 import org.junit.Test;
 
-import static org.apache.iceberg.BaseFileScanTask.SplitScanTaskIterator;
+import static org.apache.iceberg.BaseFileScanTask.FixedSizeSplitScanTaskIterator;
```

Comment:
It seems that `TestFixedSizeSplitScanTaskIterator `  and  `TestOffsetsBasedSplitScanTaskIterator` can be merged into one.

We will not have to duplicate `MockFileScanTask` and `verify` method.  In this case the verify method can take an extra `Iterator<FileScanTask>` argument.

Is this doable?

Reply:
The first argument in each `verify` method is only required to construct the Iterator, so we'd have to remove that too, if the Iterator is constructed outside

Reply:
I think it makes to sense have separate tests for the two. I moved out MockFileScanTask to an outer class, instead.

Reply:
I don't have a strong opinion on whether these should be combined. I think it's fine as long as we aren't duplicating supporting code. Sounds like that's the case now.